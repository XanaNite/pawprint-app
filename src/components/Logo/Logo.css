.logo{
  display: flex;
  color: black;
  text-decoration: none;
  text-decoration-color: black;
  text-decoration-style: none;
}

.logo img{
  width: 50px;
  height: 50px;
  margin-left: 25%;
}

.logo-title {
  width: 200px;
}
.logo-title h1{
  margin: 0 15px 0 15px;
  padding: 0;
}

.logo-title p{
  margin: 0 15px 0 15px;
  padding: 0;
}
