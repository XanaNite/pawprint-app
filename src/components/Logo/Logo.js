import React, {Component} from 'react';
import './Logo.css';
import LogoImg from '../../assets/images/pawprint-logo.png';
import { Link } from 'react-router-dom';
import TokenService from '../../services/token-service';


export default class Header extends Component {
  handleLogoutClick = () => {
    TokenService.clearAuthToken();
    window.location.reload();
  }

  render() {
    return <>
      <nav className='navigationHeader'>
        {TokenService.hasAuthToken() 
            ? <Link to='/home'>
              <div className='wholeLogo'>
              <img src={LogoImg} alt='logo' />
                <div className='logo-title'>
                  <h1>PawPrint</h1>
                  <p>Dog scheduling made easy</p>
                </div>
                </div>
             </Link>
            : <Link to='/'>              
              <div className='wholeLogo'>
              <img src={LogoImg} alt='logo' />
                <div className='logo-title'>
                  <h1>PawPrint</h1>
                  <p>Dog scheduling made easy</p>
                </div>
                </div>
            </Link>
          }
      </nav>
    </>
  };
}

//     return (
//       {TokenService.hasAuthToken() 
//         ? <Link to='/home'>{' '}Go!</Link>
//         : <Link to='/'>{' '}Go!</Link>
//       }
//       <Link to='/' className='logo'>
//           <img src={LogoImg} alt='logo' />
//             <div className='logo-title'>
//               <h1>PawPrint</h1>
//               <p>Keeping up with the office pups</p>
//             </div>
//       </Link>
//     )
// }
