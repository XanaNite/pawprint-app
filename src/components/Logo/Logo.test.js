import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Logo from './Logo';

describe(`Logo component`, () => {
  const props = {
    className: 'test-class-name',
    children: <p>test children</p>,
    'data-other': 'test-other-prop'
  };

  it('renders Logo by default', () => {
    const wrapper = shallow(<Logo />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });

  it('renders the Logo given props', () => {
    const wrapper = shallow(<Logo {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
})
