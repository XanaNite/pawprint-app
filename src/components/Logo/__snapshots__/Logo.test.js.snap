// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo component renders Logo by default 1`] = `
<NavLink
  className="logo"
  to="/"
>
  <img
    alt="logo"
    src="pawprint-logo.png"
  />
  <div
    className="logo-title"
  >
    <h1>
      PawPrint
    </h1>
    <p>
      Keeping up with the office pups
    </p>
  </div>
</NavLink>
`;

exports[`Logo component renders the Logo given props 1`] = `
<NavLink
  className="logo"
  to="/"
>
  <img
    alt="logo"
    src="pawprint-logo.png"
  />
  <div
    className="logo-title"
  >
    <h1>
      PawPrint
    </h1>
    <p>
      Keeping up with the office pups
    </p>
  </div>
</NavLink>
`;
